### PR TITLE
feat: add `--dry-run` option to fix subcommand

### PR DESCRIPTION
**Pull request by Dev380**
#### ✔ Checklist:

- [x] Commit messages have been written in [Conventional Commits](https://www.conventionalcommits.org) format;
- [x] Tests for the changes have been added (for bug fixes / features);
- [x] Docs have been added / updated on the [dotenv-linter.github.io](https://github.com/dotenv-linter/dotenv-linter.github.io) (for bug fixes / features). https://github.com/dotenv-linter/dotenv-linter.github.io/pull/61

### CODE REVIEW

**Review by mgrachev — COMMENTED** [review]
👋 Thank you for your contribution 🚀

I have left some comments. Please take a look at them 👀

**Comment by mgrachev** [comment]
I would also think about the dry-run output format, because the output for multiple files looks a bit cumbersome:
```bash
$ dotenv-linter fix --dry-run
Fixing .env
Dry run - not changing any files on disk.

DEFAULT_FROM_EMAIL="name\ example@example.ru"
FOO='#comment'

# FOO='#comment'


.env:4 ExtraBlankLine: Extra blank line detected
.env:5 ExtraBlankLine: Extra blank line detected
.env:6 ExtraBlankLine: Extra blank line detected
.env:7 ExtraBlankLine: Extra blank line detected
.env:8 DuplicatedKey: The FOO key is duplicated

Fixing .env.test
Dry run - not changing any files on disk.

FOO=#comment
TEST=nameexample@example.ru


.env.test:1 QuoteCharacter: The value has quote characters (', ")
.env.test:2 UnorderedKey: The FOO key should go before the TEST key

All warnings are fixed. Total: 7
```

@dotenv-linter/core Do you have any suggestions about this?

**Comment by DDtKey** [comment]
For dry-run it would be nice to show a formatted diff, something like that:

```diff
-abc=DEF
-ABC=DEF
A=DEF
ABC=DEF
+# ABC=DEF
+# ABC=DEF
```

File: dotenv-linter/src/common/output/fix.rs [hunk 1]
```diff
@@ -84,4 +84,17 @@ impl FixOutput {
 
         println!("{}", "Could not fix all warnings".red().bold());
     }
+
+    /// Prints dry run message
+    pub fn print_dry_run(&self, lines: Vec<LineEntry>) {
```
[the remark this answers is not in the document]
Comment (Dev380) [reply dotenv-linter/src/common/output/fix.rs]: Resolved in [2107c7d](https://github.com/dotenv-linter/dotenv-linter/pull/772/commits/2107c7d0a8fd2e6da9b15f41d2f08424bd9c5d87)****

File: dotenv-linter/src/common/output/fix.rs [hunk 2]
```diff
@@ -84,4 +84,17 @@ impl FixOutput {
...
+        if self.is_quiet_mode {
+            return;
+        }
+
+        println!("{}\n", "Dry run - not changing any files on disk.".yellow().bold());
+
+        for line in lines[..lines.len()].iter() {
```
[the remark this answers is not in the document]
Comment (Dev380) [reply dotenv-linter/src/common/output/fix.rs]: Fixed in [5d0da32](https://github.com/dotenv-linter/dotenv-linter/pull/772/commits/5d0da32931f40e28256e23ccb62857962e2ec6ba)

**Review by mgrachev — APPROVED** [review]

**Comment by mgrachev** [comment]
@Dev380 Thank you! 👍

**Pull request merged**